nix/cargo-doc-live.nix

```diff
@@ -0,0 +1,83 @@
+{ pkgs, lib, name, config, ... }:
+let
+  inherit (lib) types;
+in
+{
+  options = {
+    enable = lib.mkEnableOption name;
+
+    src = lib.mkOption {
+      type = types.path;
+      description = "The src of the cargo";
+    };
+
+    port = lib.mkOption {
+      type = types.port;
+      description = "The port for 'cargo doc'";
+      default = 8008;
+    };
+
+    crateName = lib.mkOption {
+      type = types.str;
+      description = "The crate to use when opening docs in browser";
+      default = builtins.replaceStrings [ "-" ] [ "_" ]
+        ((lib.trivial.importTOML "${config.src}/Cargo.toml").package.name);
+      defaultText = "The crate name is derived from the Cargo.toml file";
+    };
+
+    outputs.settings = lib.mkOption {
+      type = types.deferredModule;
+      internal = true;
+      readOnly = true;
+      default = {
+        processes =
+          {
```

Comment:
The `{` in line 34 is indented inconsistently with the rest of the project (e.g.: see `{` in line 32). You should fix this in similar other places as well.